catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -21,28 +21,31 @@
 import java.util.NoSuchElementException;
 import java.util.Spliterator;
 import java.util.Spliterators;
-import java.util.function.Function;
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
+import org.opengis.filter.Filter;
+import org.opengis.filter.sort.SortBy;
+
 import ddf.catalog.CatalogFramework;
 import ddf.catalog.data.Result;
 import ddf.catalog.federation.FederationException;
-import ddf.catalog.operation.QueryRequest;
 import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
 import ddf.catalog.source.SourceUnavailableException;
 import ddf.catalog.source.UnsupportedQueryException;
 
 /**
  * Effectively a cursor over the results of a filter that automatically pages through all results
  * <br/>
- * Throws a {@link CatalogCommandRuntimeException} if anything goes wrong during iteration or
+ * Throws a {@link CatalogRuntimeException} if anything goes wrong during iteration or
  * querying
  */
```

Comment:
Actually, now that I think of it, the `QueryResultIterable` _is_ thread safe, i.e., it is currently immutable and a client can use it to create as many `Iterator`s or `Stream`s as it wants and it will keep working. The `ResultQueryIterator` is actually the one that isn't thread safe.

Sooo... maybe we should remove the _Not thread safe_ comment for now. Sorry about that 😕